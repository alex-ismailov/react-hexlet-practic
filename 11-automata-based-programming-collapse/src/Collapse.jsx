/* eslint-disable jsx-a11y/anchor-is-valid */
/* eslint-disable react/static-property-placement */

import React from 'react';
import cn from 'classnames';

// BEGIN (write your solution here)
export default class Collapse extends React.Component {
  static defaultProps = {
    opened: true,
  };

  constructor(props) {
    super(props);
    this.state = {
      opened: props.opened,
    };
  }

  btnHandler = (e) => {
    e.preventDefault();
    this.setState(({ opened }) => ({ opened: !opened }));
  };

  render() {
    const { opened } = this.state;
    const { text } = this.props;

    const collapseClass = cn({
      collapse: true,
      show: opened,
    });

    return (
      <div>
        <p>
          <a className="btn btn-primary" href="#" onClick={this.btnHandler}>Link with href</a>
        </p>
        <div className={collapseClass}>
          <div className="card card-body">
            {text}
          </div>
        </div>
      </div>
    );
  }
}
// END
