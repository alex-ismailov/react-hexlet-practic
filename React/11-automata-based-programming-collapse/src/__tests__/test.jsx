import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Collapse from '../Collapse.jsx';

test('Collapse', () => {
  const vdom = <Collapse text="hi" opened={false} />;
  const wrapper = render(vdom);
  expect(wrapper.asFragment()).toMatchSnapshot();

  const button = wrapper.getByText('Link with href');

  fireEvent.click(button);
  expect(wrapper.asFragment()).toMatchSnapshot();

  fireEvent.click(button);
  expect(wrapper.asFragment()).toMatchSnapshot();

  fireEvent.click(button);
  expect(wrapper.asFragment()).toMatchSnapshot();
});

test('Collapse2', () => {
  const vdom = <Collapse text="i am grut" />;
  const wrapper = render(vdom);
  expect(wrapper.asFragment()).toMatchSnapshot();
});
