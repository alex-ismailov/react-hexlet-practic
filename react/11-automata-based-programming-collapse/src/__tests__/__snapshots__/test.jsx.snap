// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collapse 1`] = `
<DocumentFragment>
  <div>
    <p>
      <a
        class="btn btn-primary"
        href="#"
      >
        Link with href
      </a>
    </p>
    <div
      class="collapse"
    >
      <div
        class="card card-body"
      >
        hi
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Collapse 2`] = `
<DocumentFragment>
  <div>
    <p>
      <a
        class="btn btn-primary"
        href="#"
      >
        Link with href
      </a>
    </p>
    <div
      class="collapse show"
    >
      <div
        class="card card-body"
      >
        hi
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Collapse 3`] = `
<DocumentFragment>
  <div>
    <p>
      <a
        class="btn btn-primary"
        href="#"
      >
        Link with href
      </a>
    </p>
    <div
      class="collapse"
    >
      <div
        class="card card-body"
      >
        hi
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Collapse 4`] = `
<DocumentFragment>
  <div>
    <p>
      <a
        class="btn btn-primary"
        href="#"
      >
        Link with href
      </a>
    </p>
    <div
      class="collapse show"
    >
      <div
        class="card card-body"
      >
        hi
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Collapse2 1`] = `
<DocumentFragment>
  <div>
    <p>
      <a
        class="btn btn-primary"
        href="#"
      >
        Link with href
      </a>
    </p>
    <div
      class="collapse show"
    >
      <div
        class="card card-body"
      >
        i am grut
      </div>
    </div>
  </div>
</DocumentFragment>
`;
